### ISSUE TITLE
The "Skip to content" link is invalid except in the top page

### ISSUE THREAD

**Issue by jessitron** [squidfunk/mkdocs-material#899]
## Description

HtmlProofer has kindly identified that every "Skip to content" link in my site is invalid except the top-level index. You can see this on your own site: right now, the https://squidfunk.github.io/mkdocs-material/customization/ page contains the link:

```
<a href="../#customization" tabindex="1" class="md-skip">
        Skip to content
      </a>
```

and ../#customization is invalid. It should be "#customization"

### Expected behavior

All generated links on the page should be valid links.

### Actual behavior

In this line of code: https://github.com/squidfunk/mkdocs-material/blob/fa8d490b16155003386224cc82e71aafb471c828/material/base.html#L105

it does:

`href="{{ (page.toc | first).url | url }}"`

and that `url` filter turns "#internal-anchor" into "../#internal-anchor" for a top-level non-index page like Customization. For a page deeper in the docs directory it puts more `../` path components.

Unfortunately the `url` filter thinks #internal-anchors are absolute from the base, or something. I reported it to mkdocs: https://github.com/mkdocs/mkdocs/issues/1655 . There are no tests for that filter in the mkdocs repo :-(

meanwhile, you don't need to pass that url to that filter as far I can tell.

### Steps to reproduce the bug

1. Go to https://squidfunk.github.io/mkdocs-material/customization/
2. Look at the HTML
3. Search for "Skip to content"
4. Observe the invalid href: "../#customization"

Currently, your gh-pages branch says: "Deployed fa8d490 with MkDocs version: 1.0.4"

### Package versions

* Python: 3.6.0
* MkDocs: 1.0.4
* Material: 3.0.4

### Project configuration

... it shows up in your own ...

### System information

* OS: Mac
* Browser: Firefox

**Comment by squidfunk** [comment]
Thanks for reporting, that's indeed a bug! We should just remove the filter, as you already said. I'm currently on holiday, so expect a fix to land at the end of next week.

PS: that's a great issue. Would love to see more written like this as opposed to "I have error pls fix thx"

**Comment by squidfunk** [comment]
Fixed in 13a76c4. Will be part of next release. Thanks for your time in researching the problem and solution!

**Comment by squidfunk** [comment]
Released as part of 3.0.5
